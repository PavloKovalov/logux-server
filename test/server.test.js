function normalizeNewlines (string) {
  // use local copy of Jest newline normalization function
  // until Jest doens't apply normalization on comprasion
  return string.replace(/\r\n|\r/g, '\n')
}

var servers = require('./servers/servers')

Object.keys(servers).forEach(function (test) {
  it('reports ' + test, function () {
    var testCase = servers[test]()

    return testCase.then(function (result) {
      var out = result[0]
      var exit = result[1]

      if (test === 'throw' || test === 'uncatch') {
        expect(exit).toEqual(1)
      } else if (test !== 'unbind') {
        if (exit !== 0) {
          console.error(test + ' fall with:\n' + out)
        }
        expect(exit).toEqual(0)
      }
      expect(normalizeNewlines(out)).toMatchSnapshot()
    })
  })
})
