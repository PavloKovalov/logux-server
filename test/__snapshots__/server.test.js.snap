exports[`test reports destroy 1`] = `
" INFO   Logux server is listening at 1970-01-01 00:00:00        Logux server: 0.1.1        PID:          21384        Node ID:      server        Environment:  test        Subprotocol:  1.0.0        Supports:     1.x        Listen:       ws://127.0.0.1:1337

 INFO   Shutting down Logux server at 1970-01-01 00:00:00

 Custom destroy task finished
"
`;

exports[`test reports help 1`] = `
"Options:
  -h, --host  Host to bind server.                                      [string]
  -p, --port  Port to bind server                                       [number]
  -k, --key   Path to SSL key                                           [string]
  -c, --cert  Path to SSL certificate                                   [string]
  --help      Show help                                                [boolean]

Examples:
  test/servers/options.js --port 31337
  --host 127.0.0.1
  LOGUX_PORT=1337 test/servers/options.js

Corresponding ENV variables: LOGUX_HOST, LOGUX_PORT, LOGUX_KEY, LOGUX_CERT

"
`;

exports[`test reports options 1`] = `
" INFO   Logux server is listening at 1970-01-01 00:00:00        Logux server: 0.1.1        PID:          21384        Node ID:      server        Environment:  test        Subprotocol:  1.0.0        Supports:     1.x        Listen:       ws://127.0.0.1:31337

 INFO   Shutting down Logux server at 1970-01-01 00:00:00

"
`;

exports[`test reports throw 1`] = `
" ERROR  Test Error at 1970-01-01 00:00:00        fake stacktrace

 INFO   Shutting down Logux server at 1970-01-01 00:00:00

"
`;

exports[`test reports unbind 1`] = `
" INFO   Shutting down Logux server at 1970-01-01 00:00:00

"
`;

exports[`test reports uncatch 1`] = `
" ERROR  Test Error at 1970-01-01 00:00:00        fake stacktrace

 INFO   Shutting down Logux server at 1970-01-01 00:00:00

"
`;
